src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -2132,13 +2121,14 @@ internal class PSDynamicMetaObject : DynamicMetaObject
             internal PSDynamicMetaObject(Expression expression, PSObject value)
                 : base(expression, BindingRestrictions.Empty, value)
             {
+                Diagnostics.Assert(value != null, "Should be verified by callers.");
```

Comment:
Does this come from usages the ctor/class? If so please make runtime check and throw.